calico_rkt/calico_rkt.py

```diff
@@ -1,198 +1,212 @@
+
+
+def _create_calico_endpoint(container_id, netns_path, interface, subnet):
     """
-    Add a container to Calico networking with the given IP
-    Return Endpoint object
+    Configure the Calico interface for a pod.
+    Return Endpoint and IP
     """
+    _log.info('Configuring Calico networking.')
+
     try:
-        _ = client.get_endpoint(
-            hostname=hostname,
-            orchestrator_id=orchestrator_id,
-            workload_id=container_id
-        )
+        _ = _datastore_client.get_endpoint(hostname=HOSTNAME,
+                                           orchestrator_id=ORCHESTRATOR_ID,
+                                           workload_id=container_id)
     except KeyError:
         # Calico doesn't know about this container.  Continue.
         pass
     else:
-        print_stderr("This container has already been configured with Calico Networking.")
+        _log.error(
+            "This container has already been configured with Calico Networking.")
         sys.exit(1)
 
-    # Allocate and Assign ip address through IPAM Client
-    ip = _allocate_IP()
+    endpoint, ip = _container_add(hostname=HOSTNAME,
+                                  orchestrator_id=ORCHESTRATOR_ID,
+                                  container_id=container_id,
+                                  netns_path=netns_path,
+                                  interface=interface,
+                                  subnet=subnet)
+
+    _log.info('Finished configuring network interface')
+    return endpoint, ip
+
+
+def _container_add(hostname, orchestrator_id, container_id, netns_path, interface, subnet):
+    """
+    Add a container to Calico networking
+    Return Endpoint object and newly allocated IP
+    """
+    # Allocate and Assign ip address through IPAM _datastore_client
+    pool = _generate_pool(subnet)
+    ip = _allocate_ip(pool)
 
     # Create Endpoint object
     try:
-        ep = client.create_endpoint(HOSTNAME, ORCHESTRATOR_ID,
-                                      container_id, [IPAddress(ip)])
+        ep = _datastore_client.create_endpoint(HOSTNAME, ORCHESTRATOR_ID,
+                                               container_id, [ip])
     except AddrFormatError:
-        print_stderr("This node is not configured for IPv%d. Unassigning IP "\
-                      "address %s then exiting."  % ip.version, ip)
-        client.unassign_address(IPNetwork(input_json['ipam']['subnet']), ip)
+        _log.error("This node is not configured for IPv%d. Unassigning IP "
+                   "address %s then exiting." % ip.version, ip)
+        _datastore_client.unassign_address(pool, ip)
         sys.exit(1)
 
     # Create the veth, move into the container namespace, add the IP and
     # set up the default routes.
     ep.mac = ep.provision_veth(Namespace(netns_path), interface)
-    client.set_endpoint(ep)
+    _datastore_client.set_endpoint(ep)
 
-    # Let the caller know what endpoint was created.
     return ep, ip
```

Comment:
I'm wondering if this is necessary, since IP is stored in ep.ip_nets. But this is just a note-to-self, not sure if action is required